client/src/main/java/org/eclipse/hono/client/impl/AbstractHonoClient.java

```diff
@@ -221,4 +222,49 @@ protected static final void setApplicationProperties(final Message msg, final Ma
             msg.setApplicationProperties(applicationProperties);
         }
     }
+
+    /**
+     * TODO.
+     */
+    protected final void startAutoCloseTimer() {
+        final long inactiveLinkTimeout = connection.getConfig().getInactiveLinkTimeout();
+        if (inactiveLinkTimeout > 0) {
+            startAutoCloseTimer(inactiveLinkTimeout);
+        }
+    }
+
+    /**
+     * Stores the current time stamp on the sender link. This is used to detect and close inactive AMQP sender links.
+     */
+    protected final void storeLastSendTime() {
+        sender.attachments().set("last-send-time", Long.class, Instant.now().toEpochMilli());
+    }
+
+    private void startAutoCloseTimer(final long delay) {
+        connection.getVertx().setTimer(delay, id -> {
+            final Long lastSendTime = sender.attachments().get("last-send-time", Long.class);
+            long remaining = 0;
+            if (lastSendTime != null) {
+                remaining = getRemainingTimeout(lastSendTime, Instant.now().toEpochMilli(),
+                        connection.getConfig().getInactiveLinkTimeout());
+            }
+
+            if (lastSendTime == null || remaining == 0) {
+                closeLinks(v -> {
+                });
+            } else {
+                startAutoCloseTimer(remaining);
+            }
+        });
+    }
+
+    // visible for testing
+    long getRemainingTimeout(final long lastSend, final long now, final long timeout) {
+        final long timeDiff = now - lastSend;
+        if (timeDiff >= timeout) {
+            return 0L;
+        } else {
+            return timeout - timeDiff;
```

Comment:
I think we should be a little more lenient here:
```
} else {
  final long remainingTimeout = timeout - timeDiff;
  if (remainingTimeout > timeDiff) {
    return timeout;
  } else {
    return remainingTimeout;
  }
}
```
otherwise we might end up with starting a new timer (with ever decreasing delay) for each message being sent, right?

Reply:
Note: @sophokles73 and I agreed offline that this is not an issue.